invenio_oauthclient/contrib/cern.py

```diff
@@ -212,13 +212,34 @@ def account_setup(remote, token):
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
+        groups = fetch_groups(res['Group'])
+        current_user.info['group'] = groups
         current_user.modified = True
         current_user.save()
 
-        if user and not any([user.family_name, user.given_names]):
-            user.family_name = res['Lastname'][0]
-            user.given_names = res['Firstname'][0]
-
-            db.session.add(user)
-            current_user.reload()
+        # update user object
+        if user:
+            modified = False
+
+            # try to add groups
+            try:
+                from invenio_groups.models import Group
+                for groupname in groups:
+                    candidates = Group.query.filter(Group.name == groupname).all()
+                    for candidate in candidates:
+                        candidate.add_member(user)
+                        db.session.add(candidate)
+                        modified = True
+            except ImportError:
+                # no invenio-groups installed
```

Comment:
Can you register a warning in this case, so that it's clear for the admin why groups are not updated? 
I'm not sure how we will use the invenio-groups module - I don't think we are using the groups now, but it's good to have it in the logs if anything is expected, but it's not installed.
One more thing: how do you remove people from groups in this case?